fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java

```diff
@@ -989,7 +989,8 @@ public TExpr treeToThrift() {
             // Hack to ensure BE never sees TYPE_NULL. If an expr makes it this far without
             // being cast to a non-NULL type, the type doesn't matter and we can cast it
             // arbitrarily.
-            Preconditions.checkState(this instanceof NullLiteral || this instanceof SlotRef);
+            Preconditions.checkState(this instanceof NullLiteral || this instanceof SlotRef
+                    || this instanceof VariableExpr);
```

Comment:
why we could see VariableExpr here?

Reply:
Because when we `set @a=null` and use the `@a` in the statement, for example, `select @a ....`. The `@a` will be built as `VariableExpr` not the `NullLiteral`